src/battle/anim/ghost.c

```diff
@@ -189,3 +202,402 @@ const struct SpriteTemplate gSpriteTemplate_83DAF98 =
+                "\torrs r0, r2\n"
+                "\tstrb r0, [r4]\n"
+                "_080DE13E:\n"
+                "\tmovs r1, 0x2E\n"
+                "\tldrsh r0, [r3, r1]\n"
+                "\tcmp r0, 0\n"
+                "\tbeq _080DE14E\n"
+                "\tcmp r0, 0x1\n"
+                "\tbeq _080DE158\n"
+                "\tmovs r6, 0x1\n"
+                "\tb _080DE162\n"
+                "_080DE14E:\n"
+                "\tmovs r1, 0x30\n"
+                "\tldrsh r0, [r3, r1]\n"
+                "\tcmp r0, 0x2\n"
+                "\tbne _080DE162\n"
+                "\tb _080DE166\n"
+                "_080DE158:\n"
+                "\tmovs r1, 0x30\n"
+                "\tldrsh r0, [r3, r1]\n"
+                "\tcmp r0, 0x4\n"
+                "\tbne _080DE162\n"
+                "\tmovs r5, 0x1\n"
+                "_080DE162:\n"
+                "\tcmp r5, 0\n"
+                "\tbeq _080DE198\n"
+                "_080DE166:\n"
+                "\tldrb r2, [r4]\n"
+                "\tlsls r0, r2, 29\n"
+                "\tlsrs r0, 31\n"
+                "\tmovs r1, 0x1\n"
+                "\teors r1, r0\n"
+                "\tlsls r1, 2\n"
+                "\tmovs r0, 0x5\n"
+                "\tnegs r0, r0\n"
+                "\tands r0, r2\n"
+                "\torrs r0, r1\n"
+                "\tstrb r0, [r4]\n"
+                "\tldrh r0, [r3, 0x32]\n"
+                "\tadds r0, 0x1\n"
+                "\tmovs r1, 0\n"
+                "\tstrh r0, [r3, 0x32]\n"
+                "\tstrh r1, [r3, 0x30]\n"
+                "\tlsls r0, 16\n"
+                "\tasrs r0, 16\n"
+                "\tcmp r0, 0x5\n"
+                "\tbne _080DE1AA\n"
+                "\tstrh r1, [r3, 0x32]\n"
+                "\tldrh r0, [r3, 0x2E]\n"
+                "\tadds r0, 0x1\n"
+                "\tstrh r0, [r3, 0x2E]\n"
+                "\tb _080DE1AA\n"
+                "_080DE198:\n"
+                "\tcmp r6, 0\n"
+                "\tbeq _080DE1A4\n"
+                "\tadds r0, r3, 0\n"
+                "\tbl DestroyAnimSprite\n"
+                "\tb _080DE1AA\n"
+                "_080DE1A4:\n"
+                "\tldrh r0, [r3, 0x30]\n"
+                "\tadds r0, 0x1\n"
+                "\tstrh r0, [r3, 0x30]\n"
+                "_080DE1AA:\n"
+                "\tpop {r4-r6}\n"
+                "\tpop {r0}\n"
+                "\tbx r0\n");
+}
+
+void sub_80DE1B0(u8 taskId) {
+    struct Task *task;
+    
+    task = &gTasks[taskId];
+    task->data[0] = duplicate_obj_of_side_rel2move_in_transparent_mode(1);
+    if (task->data[0] < 0) {
+        DestroyAnimVisualTask(taskId);
+        return;
+    }
+    task->data[1] = 0;
+    task->data[2] = 15;
+    task->data[3] = 2;
+    task->data[4] = 0;
+    REG_BLDCNT = 0x3F40;
+    REG_BLDALPHA = (task->data[3] << 8) | task->data[2];
+    gSprites[task->data[0]].data[0] = 80;
+    if (GetBattlerSide(gAnimBankTarget) == 0) {
+        gSprites[task->data[0]].data[1] = 0xff70;
```

Comment:
`0xff70` should be `-144`.  The `sprite.data` array is `s16`.  Same with the `else` block right below.